### ISSUE TITLE
Evaluate Visual-Grounding pre-training results

### ISSUE THREAD

**Issue by hanochk** [OFA-Sys/OFA#104]
Hi, trying to evaluate pre-trained large best OFA with VG challenge and when running the 1st part of evaluate_refcoco.sh (attached the pasted part), results were given into json file in the following format in contrast to what described in the readme:

[{"uniq_id": "3641_0", "box": [353.07916259765625, 0.0, 501.7441101074219, 370.531494140625]}, {"uniq_id": "3641_1", "box": [258.88201904296875, 0.0, 501.7441101074219, 414.5508117675781]}, {"uniq_id": "3642_0", "box": [461.3739013671875, 17.094879150390625, 565.1830444335938, 357.7103576660156]}, {"uniq_id": "3642_1", "box": [554.2894287109375, 0.0, 640.1563110351562, 400.8749084472656]}, {"uniq_id": "3642_2", "box": [453.0435485839844, 17.094879150390625, 565.1830444335938, 357.7103576660156]},

i.e no captions inside as was described!

I'm trying to visualize the results over the refCOCO images together with the captions, as you made in the paper, tried with the following evaluate_caption.sh that accepts the annotation, json file. So i gave
the following  arguments, where "refcoco_val_predict.json" is the results of the VG as before:

python coco_eval.py ../../results/caption/test_predict.json ../../results/refcoco_val_predict.json
  
Can you enlight me what am I doing wrong? what should be the right way to plot the refCOCO images with the VGresults over it?

#user_dir=../../ofa_module
user_dir=../../models/ofa/ofa_module
bpe_dir=../../utils/BPE
selected_cols=0,4,2,3

data=../../dataset/refcoco_data/refcoco_val.tsv
path=../../checkpoints/refcoco_large_best.pt
result_path=../../results/refcoco
split='refcoco_val'
python3 -m torch.distributed.launch --nproc_per_node=${GPUS_PER_NODE} --master_port=${MASTER_PORT} ../../evaluate.py \
    ${data} \
    --path=${path} \
    --user-dir=${user_dir} \
    --task=refcoco \
    --batch-size=16 \
    --log-format=simple --log-interval=10 \
    --seed=7 \
    --gen-subset=${split} \
    --results-path=${result_path} \
    --beam=5 \
    --min-len=4 \
    --max-len-a=0 \
    --max-len-b=4 \
    --no-repeat-ngram-size=3 \
    --fp16 \
    --num-workers=0 \
    --model-overrides="{\"data\":\"${data}\",\"bpe_dir\":\"${bpe_dir}\",\"selected_cols\":\"${selected_cols}\"}"

exit 0

**Comment by logicwong** [comment]
@hanochk Hi, You can find the corresponding caption and image from the original file (i.e., refcoco_val.tsv) according to 'uniq_id' in the result. As for the visualization, you can refer to our [colab](https://colab.research.google.com/drive/1AHQNRdaUpRTgr3XySHSlba8aXwBAjwPB?usp=sharing)

**Comment by logicwong** [comment]
@hanochk 

```
src_tokens = sample['net_input']['src_tokens'][0]
src_tokens = utils.strip_pad(src_tokens, task.tgt_dict.pad())
caption = task.bpe.decode(task.tgt_dict.string(src_tokens))
```

**Comment by hanochk** [comment]
How may the Pretraining model OFA-Large for instance could be used for one of the tasks such as visual grounding? It can't infer the relevant task out of the ofa_large.pt model in :

    models, saved_cfg, task = checkpoint_utils.load_model_ensemble_and_task(
        utils.split_paths(cfg.common_eval.path),
        arg_overrides=overrides,
        suffix=cfg.checkpoint.checkpoint_suffix,
        strict=(cfg.checkpoint.checkpoint_shard_count == 1),
        num_shards=cfg.checkpoint.checkpoint_shard_count,

**Comment by logicwong** [comment]
duplicated of #109
